subsys/bluetooth/audio/capabilities.c

```diff
@@ -189,9 +178,7 @@ static int unicast_server_metadata_cb(struct bt_audio_stream *stream,
 {
 	struct bt_audio_capability *cap = stream->user_data;
 
-	if (cap->ops == NULL || cap->ops->metadata == NULL) {
-		return -EACCES;
-	}
+	__ASSERT_NO_MSG(cap->ops->metadata);
```

Comment:
If we want to mandate that each of these are assigned (which isn't a bad idea), we should probably ensure that the documentation also reflects that.